pkg/api/controller.go

```diff
@@ -3006,6 +3011,72 @@ func (c *Controller) GetCurrentUser(w http.ResponseWriter, r *http.Request) {
 	writeResponse(w, http.StatusOK, response)
 }
 
+func (c *Controller) sendResetPasswordEmail(email []string, token string) error {
+	err := c.Emailer.SendEmail(email, token, token, []string{})
+	if err != nil {
+		c.Logger.
+			WithError(err).
+			WithFields(logging.Fields{
+				"email": email,
+			}).Debug("Error sending email")
+		return err
+	}
+	c.Logger.WithFields(logging.Fields{
+		"email": email,
+	}).Info("Successfully sent email")
+	return err
+}
+
+func (c *Controller) PasswordForgot(w http.ResponseWriter, r *http.Request, body PasswordForgotJSONRequestBody) {
+	user, err := c.Auth.GetUserByEmail(r.Context(), body.Email)
+	if err != nil {
+		writeError(w, http.StatusUnauthorized, http.StatusText(http.StatusUnauthorized))
+		return
+	}
+	if *user.Email != body.Email {
+		writeError(w, http.StatusUnauthorized, http.StatusText(http.StatusUnauthorized))
+
+		return
+	}
+	secret := c.Auth.SecretStore().SharedSecret()
+	currentTime := time.Now()
+	token, err := GenerateJWTResetPassword(secret, ResetPasswordAudience, user.ID, body.Email, currentTime, currentTime.Add(tokenExpiryDuration))
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err)
+		return
+	}
+	// TODO (@shimi9276) create template for sending the email with link for reset
+	err = c.sendResetPasswordEmail([]string{body.Email}, token)
+	if err != nil {
+		writeError(w, http.StatusBadRequest, err)
+		return
+	}
+	writeResponse(w, http.StatusOK, err)
+}
+
+func (c *Controller) UpdatePassword(w http.ResponseWriter, r *http.Request, body UpdatePasswordJSONRequestBody) {
+	claims, err := VerifyResetPasswordToken(c.Auth, body.Token)
+	if err != nil {
+		writeError(w, http.StatusUnauthorized, err)
+		return
+	}
+	user, err := c.Auth.GetUserByEmail(r.Context(), claims.Id)
```

Comment:
here we will need to pull the email from a different field (not Id) - related to another comment in this review

Reply:
Fixed